app/code/Magento/Newsletter/Model/Subscriber.php

```diff
@@ -604,20 +636,14 @@ protected function _updateCustomerSubscription($customerId, $subscribe)
 
         $this->save();
         $sendSubscription = $sendInformationEmail;
-        if ($sendSubscription === null xor $sendSubscription && $this->isStatusChanged()) {
+        if ($sendSubscription === null xor $sendSubscription) {
             try {
-                switch ($status) {
-                    case self::STATUS_UNSUBSCRIBED:
-                        $this->sendUnsubscriptionEmail();
-                        break;
-                    case self::STATUS_SUBSCRIBED:
-                        $this->sendConfirmationSuccessEmail();
-                        break;
-                    case self::STATUS_NOT_ACTIVE:
-                        if ($isConfirmNeed) {
-                            $this->sendConfirmationRequestEmail();
-                        }
-                        break;
+                if ($isConfirmNeed) {
```

Comment:
Here you dropped check for `$this->isStatusChanged()` and `$status !== self::STATUS_NOT_ACTIVE`. Why  it's needed?